pkg/publish/options.go

```diff
@@ -105,6 +107,17 @@ func WithTagOnly(tagOnly bool) Option {
 func Insecure(b bool) Option {
 	return func(i *defaultOpener) error {
 		i.insecure = b
+		t, ok := i.t.(*http.Transport)
+		if !ok {
+			return fmt.Errorf("unable to configure insecure roundtripper (not HTTP)")
```

Comment:
I don't think it makes sense to error here.

If we can't override the TLSClientConfig, then in the case where we needed to, it will fail anyway, and in the case where we _didn't_ need to, it fails even when it shouldn't.

Reply:
I've switched this -- do you think I should log anything here for the debugging case?

Reply:
Logging sounds reasonable, but also the only time anyone is going to hit this is if they are using `ko` as a library _and_ supplying their own transport, so it might not really matter.